github/interactions_repos.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// GetRestrictions fetches the interaction restrictions for a repository.
+//
+// GitHub API docs: https://developer.github.com/v3/interactions/repos/#get-interaction-restrictions-for-a-repository
+func (s *InteractionsService) GetRestrictions(ctx context.Context, owner, repo string) (*InteractionRestriction, *Response, error) {
```

Comment:
@gmlewis I was wondering if I should rename `GetRestrictions` to `GetRestrictionsForRepo` so that for Organization interactions API, we can name methods as `GetRestrictionsForOrg`?

Reply:
That sounds like a good idea, @waseem18!